test/robustness/overlay/areal_areal/random_integer_grids.cpp

```diff
@@ -0,0 +1,267 @@
+        success = false;
+    }
+}
+
+bool test_all(grid_settings const& settings)
+{
+    prng gen(settings.seed);
+    const auto grid = grid_cells(settings);
+    const auto points = test_points(settings);
+    bool all_success = true;
+    for (int i = 0; i < settings.count || settings.count == -1; i++)
+    {
+        const bits bits1 = gen_bits(gen, settings.width * settings.height);
+        const bits bits2 = gen_bits(gen, settings.width * settings.height);
+        const auto geo1 = bits_to_geometry(bits1, grid, points, settings, all_success);
+        const auto geo2 = bits_to_geometry(bits2, grid, points, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "union", std::bit_or<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::union_(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "intersection", std::bit_and<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::intersection(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "sym_difference", std::bit_xor<>{},
+                [](mp const& g1, mp const& g2) { mp g; bg::sym_difference(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "difference g1 \\ g2", 
+                [](std::bitset<chunk_size> b1, std::bitset<chunk_size> b2) { return b1 & (~b2); },
+                [](mp const& g1, mp const& g2) { mp g; bg::difference(g1, g2, g); return g; },
+                points, grid, settings, all_success);
+        test_op(bits1, bits2, geo1, geo2, "difference g2 \\ g1", 
+                [](std::bitset<chunk_size> b1, std::bitset<chunk_size> b2) { return b2 & (~b1); },
+                [](mp const& g1, mp const& g2) { mp g; bg::difference(g2, g1, g); return g; },
+                points, grid, settings, all_success);
+    }
+    return all_success;
+}
+
+int main(int argc, char** argv)
+{
+    BoostGeometryWriteTestConfiguration();
+    try
+    {
+        namespace po = boost::program_options;
+        po::options_description description("=== random_integer_grids ===\nAllowed options");
+
+        grid_settings settings;
+
+        description.add_options()
+            ("help", "Help message")
+            ("seed",
+             po::value<decltype(settings.seed)>(&settings.seed)->default_value(settings.seed),
+             "Initialization seed for random generator")
+            ("count",
+             po::value<decltype(settings.count)>(&settings.count)->default_value(settings.count),
+             "Number of tests (-1 for infinite loop)")
+            ("width",
+             po::value<decltype(settings.width)>(&settings.width)->default_value(settings.width),
+             "Width of grid")
+            ("height",
+             po::value<decltype(settings.height)>(&settings.height)->default_value(settings.height),
```

Comment:
The decltype is elegant. But because it's an `int`, if I specify `-10` for height, I get a segmentation fault...